### DISCUSSION TITLE
possibility using this library from IronPython

### DISCUSSION THREAD

**Discussion by xyont** [BriefFiniteElementNet/BriefFiniteElement.Net#162] [General]
hi,

an interesting library to use, i'm trying but not succeed in solving steps. 

some error messages shown,
`IOError: [Errno 2] Could not load file or assembly 'System.Runtime.CompilerServices.Unsafe, Version=4.0.4.1, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.`

any hints or suggestion?

regards,

![2023-10-15 19_03_00-test26 py - SciTE](https://github.com/BriefFiniteElementNet/BriefFiniteElement.Net/assets/72367311/8ab0c4ba-5d84-4821-9fe1-9008e5b3d43b)

**Comment by epsi1on** [comment]
Hi,
Not sure, but actually this library do not need C# unsafe memory access. 
It is interesting that line 30 of your code throws the error, which means there is only problem with `Model.Solve()` method.
Can you please also check the `Model.Solve_MPC()` method too?

**Comment by xyont** [comment]
indeed, all model definition is work except in `model.Solve()`. I guessed this problem's related to `CSparse.dll` and `System.Memory.dll` incompatibilities. Where's another source to download of both DLL's file instead from NuGet?

also, i'm checking with `model.Solve_MPC()` but the problem persists.

**edited
previously i'm missing some DLL and trying to repair by add, however another problem in `CSparse.dll` and shown error message as:

`SystemError: Method 'Solve' in type 'CSparse.Double.Factorization.DenseLU' from assembly 'BriefFiniteElementNet.Common, Version=2.0.5.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.`

thanks in advance,

![2023-10-16 04_22_25-test26 py - SciTE](https://github.com/BriefFiniteElementNet/BriefFiniteElement.Net/assets/72367311/a725b44e-e401-482c-8842-4a2abc216739)

**Reply by epsi1on** [reply]
Could you please check and tell me the version of `CSparse.dll` which you are referencing in the code (line 3)?
you should use V3.5 ([here is nuget](https://www.nuget.org/packages/CSparse/3.5.0)). because the later version after 3.5 are not compatible with dotnet4.5 (they are dotnet 4.6 and later) BFE stuck with Csparse V 3.5 in order to be compatible with net45

**Reply by xyont** [reply]
many thanks for the hints about `CSparse.dll` incompatibilities,  i do roll-back to previous version as suggested and it  seems to be working now.

i'll try to use the library from IronPython and reporting when still have any issues.

**Comment by epsi1on** [comment]
@xyont could we please  continue this conversation in issue #165 ? 
Thanks

**Comment by epsi1on** [comment]
inactivity
